### ISSUE TITLE
SS5: Better PSR-2 conformance

### ISSUE THREAD

**Issue by tractorcow** [silverstripe/silverstripe-framework#7991]
## Affected Version

5.x-dev

## Description

A set of PSR-2 rules are ignored for framework, specifically because fixing them would require breaking changes to code. These exclusions are necessary for 4.x but should all be fixed in 5.x.

## Steps to Reproduce

```
git clone https://github.com/silverstripe/silverstripe-framework.git -b=master
cd silverstripe-framework
phpcs --standard=PSR2 ./src/ ./tests/
```

many errors :)

## Todo-list

Eliminate all PSR* exclusions in https://github.com/silverstripe/silverstripe-framework/blob/4.0/phpcs.xml.dist for master branch.

**Comment by robbieaverill** [comment]
I've made a start at #7993

**Comment by tractorcow** [comment]
I've been doing quite a lot of class renaming, getting rid of all classes with `_` in them, as well as `SS` prefixed classes.

**Comment by tractorcow** [comment]
Branches:

 - [ ] https://github.com/open-sausages/silverstripe-framework/tree/pulls/5.0/psr2-class-renaming

**Comment by sminnee** [comment]
Yeah, PSR-12, Slevomat, PHP 7.2+ or even PHP 7.3+ support would get a :+1: from me in principle.

**Comment by robbieaverill** [comment]
The good news is that a lot of the slevomat standards are "fixable" with phpcbf, e.g. PHP return type hinting etc

**Comment by robbieaverill** [comment]
Hey @tractorcow, with context of your WIP branches above and [RFC: Rename all classes with underscores to be PSR-2 compliant](https://github.com/silverstripe/silverstripe-framework/issues/8636), do you want to revive those branches and make PRs for them? Or should whoever picks up #8636 pick your branches up and roll with those?

**Comment by robbieaverill** [comment]
I've changed this to rfc/accepted, with me @sminnee and @tractorcow in support. I find it unlikely that the rest of the core team would be against this as well

**Comment by tractorcow** [comment]
Sorry @robbieaverill I'm likely not going to be able to contribute the time needed to do such a large upgrade sorry. =( I suggest someone use my branches as inspiration, and just repeat it on the latest branches.

**Comment by dhensby** [comment]
I'm happy for us to aim for PSR-12 compliance but it shouldn't be done at any cost. I can see that the PSR-12 standard is much more detailed and prescriptive than the previous standards.

**Comment by sminnee** [comment]
Rule of thumb: that any API changes introduced by adopting the PSR should be handled by silverstripe-upgrader. If silverstripe-upgrader can't reliably do this, then the work's not likely to be worth it.
